src/relayclient/GSNConfigurator.ts

```diff
@@ -0,0 +1,132 @@
+import { Address, IntString } from './types/Aliases'
+import { defaultEnvironment } from './types/Environments'
+
+const GAS_PRICE_PERCENT = 20
+const MAX_RELAY_NONCE_GAP = 3
+const DEFAULT_RELAY_TIMEOUT_GRACE_SEC = 1800
+
+const defaultGsnConfig: GSNConfig = {
+  contractInteractorConfig: {
+    gtxdatanonzero: defaultEnvironment.gtxdatanonzero,
+    verbose: false
+  },
+  relayClientConfig: {
+    gasPriceFactorPercent: GAS_PRICE_PERCENT,
+    minGasPrice: '0',
+    maxRelayNonceGap: MAX_RELAY_NONCE_GAP,
+    verbose: false
+  },
+  knownRelaysManagerConfig: {
+    relayTimeoutGrace: DEFAULT_RELAY_TIMEOUT_GRACE_SEC,
+    verbose: false
+  },
+  transactionValidatorConfig: {
+    verbose: false
+  },
+  relayProviderConfig: {
+    verbose: false
+  },
+  accountManagerConfig: {
+    verbose: false,
+    methodSuffix: '',
+    jsonStringifyRequest: false
+  },
+  chainId: defaultEnvironment.chainId,
+  relayHubAddress: '0x0000000000000000000000000000000000000000'
+}
+
+/**
+ * All classes in GSN must be configured correctly with non-null values.
+ * Yet it is tedious to provide default values to all configuration fields on new instance creation.
+ * This helper allows users to provide only the overrides and the remainder of values will be set automatically.
+ */
+export function configureGSN (partialConfig: RecursivePartial<GSNConfig>): GSNConfig {
+  return mergeDeep({}, defaultGsnConfig, partialConfig) as GSNConfig
+}
+
+type RecursivePartial<T> = {
+  [P in keyof T]?: RecursivePartial<T[P]>;
+}
+
+/**
+ * Simple object check.
+ * @param item
+ * @returns {boolean}
+ */
+function isObject (item: any): boolean {
+  return (item != null && typeof item === 'object' && !Array.isArray(item))
+}
+
+/**
+ * Deep merge two objects.
+ * @param target
+ * @param sources
+ */
+function mergeDeep (target: any, ...sources: any[]): Object {
```

Comment:
I don't want to add dependencies for this one place I need it. 
It will be marginally useful elsewhere, deep merge is considered a dangerous thing to do. I can do it here because I know I am merging very primitive objects

Reply:
still, its an external utility, and also deserves a test. or maybe use external tool, like `deepmerge` ?